src/index.ts

```diff
@@ -0,0 +1,134 @@
+import * as net from 'node:net';
+import { Logger } from './Logger';
+import { reserve as reserveTcp } from './tcp';
+import { reserve as reserveUdp } from './udp';
+
+const logger = new Logger();
+
+type Type = 'udp' | 'tcp';
+
+// Store picked ports for the specified reserveTimeout time.
+// This Set stores strings/hashes with the form "type:ip:port".
+const reserved: Set<string> = new Set();
+
+export async function pickPort(
+	{
+		type = 'udp',
+		ip = '0.0.0.0',
+		minPort = 10000,
+		maxPort = 20000,
+		reserveTimeout = 5
+	}:
+	{
+		type?: Type;
+		ip?: string;
+		minPort?: number;
+		maxPort?: number;
+		reserveTimeout?: number;
+	}	= {}
+): Promise<number>
+{
+	logger.debug(
+		`pickPort() [type:${type}, ip:${ip}, minPort:${minPort}, maxPort:${maxPort}, reserveTimeout:${reserveTimeout}]`
+	);
+
+	// Sanity checks.
+
+	type = type.toLowerCase() as Type;
+
+	const family = net.isIP(ip);
+
+	if (type !== 'udp' && type !== 'tcp')
+	{
+		throw new TypeError('invalid type parameter');
+	}
```

Comment:
Let's rely on the type system if we are using Typescript. We don't need to manually check the argument types.

Reply:
We have talked about this in the past (mediasoup related but same topic). The fact that we write a lib in TypeScript doesn't mean that all users use TypeScript. They may use JavaScript so TS checking will not happen. Said that: do we want to also protect those users with runtime checks or not?

Reply:
I think we do not have to carry with that responsibility. Having said that, it's an extra protection for JS usages, so OK, let's keep it.